src/relay/pass/type_solver.h

```diff
@@ -131,6 +136,8 @@ class TypeSolver {
   common::Arena arena_;
   /*! \brief Reporter that reports back to self */
   TypeReporter reporter_;
+  /*! \brief Data structure for unifying types */
+  TypeUnifier unifier_;
```

Comment:
Do we need to expose TypeUnifier? Or can we hide it behind the  TypeSolver?

Reply:
I think we should be able to merge with the solver, there should be a single map from type vars.